cleverhans/attacks.py

```diff
@@ -429,7 +430,20 @@ def generate(self, x, **kwargs):
             'clip_max': self.clip_max
         }
 
-        FGM = FastGradientMethod(self.model, back=self.back, sess=self.sess)
```

Comment:
I think it's better not to modify this class to test for TFE, my original idea was just to define the fast gradient method to use as a constant attached to class, for example:
```
class BasicIterativeMethod(Attack):
    FGM_CLASS = FastGradientMethod

    ...
    def generate(self, x, **kwargs):
        ...
        FGM = self.FGM_CLASS(self.model, back=self.back, sess=self.sess)
        ...
 ```

This way in TFE when you create your new TFE specific class you overwrite the constant with the proper TFE FGM class.
```
# In attacks_tfe.py
class FastGradientMethod(Attack, attacks.FastGradientMethod):
    FGM_CLASS = FastGradientMethod  # It will pick the one local to your file.
```

This way you don't have to rewrite the method for generate and the code changes are minimal.

Reply:
Got it!